mindsdb/libs/phases/data_extractor/data_extractor.py

```diff
@@ -70,20 +70,25 @@ def _get_prepared_input_df(self):
         if 'from_data' in self.transaction.hmd and self.transaction.hmd['from_data'] is not None:
             # make sure we build a dataframe that has all the columns we need
             df = self.transaction.hmd['from_data']
-            df = df.where((pandas.notnull(df)), None)
+            # @TODO This gets converted to a pandas Datafram from a mindsdb DataSource , not sure why
+            df = df.where((pd.notnull(df)), None)
 
         # if this is a predict statement, create use model_when_conditions to shape the dataframe
         if  self.transaction.lmd['type'] == TRANSACTION_PREDICT:
             if self.transaction.hmd['when_data'] is not None:
                 df = self.transaction.hmd['when_data']
-                df = df.where((pandas.notnull(df)), None)
+                # @TODO This gets converted to a pandas Datafram from a mindsdb DataSource here, not sure why
+                df = df.where((pd.notnull(df)), None)
```

Comment:
we need to take out the null rows

Reply:
no change was made here other than renaming `pd` to `pandas`